enhancements/scheduling/descheduler-profiles.md

```diff
@@ -0,0 +1,259 @@
+to the user.
+
+The profiles will be translated to an upstream Descheduler policy enabling them:
+
+* `AffinityAndTaints`:
+```yaml
+apiVersion: "descheduler/v1alpha1"
+kind: "DeschedulerPolicy"
+strategies:
+  "RemovePodsViolatingInterPodAntiAffinity":
+    enabled: true
+  "RemovePodsViolatingNodeTaints":
+    enabled: true
+  "RemovePodsViolatingNodeAffinity":
+    enabled: true
+    params:
+      nodeAffinityType:
+      - "requiredDuringSchedulingIgnoredDuringExecution"
+```
+
+* `TopologyAndDuplicates`:
+```yaml
+apiVersion: "descheduler/v1alpha1"
+kind: "DeschedulerPolicy"
+strategies:
+  "RemovePodsViolatingTopologySpreadConstraint":
+    enabled: true
+  "RemoveDuplicates":
+    enabled: true
+```
+
+* `LifecycleAndUtilization`:
+```yaml
+apiVersion: "descheduler/v1alpha1"
+kind: "DeschedulerPolicy"
+strategies:
+  "PodLifeTime":
+     enabled: true
+     params:
+       podLifeTime:
+         maxPodLifeTimeSeconds: 86400 #24 hours
+  "RemovePodsHavingTooManyRestarts":
+     enabled: true
+     params:
+       podsHavingTooManyRestarts:
+         podRestartThreshold: 100
+         includingInitContainers: true
+  "LowNodeUtilization":
+     enabled: true
+     params:
+       nodeResourceUtilizationThresholds:
+         thresholds:
+           "cpu" : 20
+           "memory": 20
+           "pods": 20
+         targetThresholds:
+           "cpu" : 50
+           "memory": 50
+           "pods": 50
+```
+
+**Note:** The `LifecycleAndAutomation` profile contains those strategies which have the
+most available parameters for users to tweak, and we must decide on sensible default values
+for these parameters (the values above are taken from the upstream Descheduler readme).
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+The test plan should remain similar to what is currently in place for the Descheduler + Operator.
+Ensuring that the operator spec settings are correctly translated to a policy file that is used
+by the descheduler will remain the same.
+
+This may complicate how we test individual strategies, as while they are grouped it will be tougher
+to distinctly get the expected outcome from a pod that is evictable by multiple strategies (for example,
+a test environment designed to get evictions for LowUtilization may also have some pods older than PodLifeTime).
+These strategies won't conflict with each other per se, they will only make setting up a specific
+test environment more difficult. One option to mitigate this would be breaking up the profiles further, or even
+grouping some strategies into their own profiles.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Ideally, this field will immediately replace the existing `strategies` config as the initial GA option for
+configuring the descheduler.
+
+However, since the current API is version `v1beta1` we may be required to continue supporting the
```

Comment:
I'd clearly express the deprecation timeline, which is Beta: 9 months or 3 releases (whichever is longer), for beta-level APIs. For conversions you'll need to have both (`strategies` and `profile`) added to both APIs, but v1beta1 version should be deprecated right away, same with `strategies` field in v1, and point to using the new `profiles`.

Reply:
ok, so for the v1 API we should still provide strategies with a note that it's deprecated and will be removed? Or just provide the v1 API with only `profiles` and still support the v1beta1 api with `strategies` until that API is removed according to that timeline?

Reply:
@damemi yeah, for v1 provide deprecated strategies. We should also announce deprecation of v1beta1 too.